src/Clearinghouse.sol

```diff
@@ -389,6 +389,15 @@ contract Clearinghouse is Policy, RolesConsumer, CoolerCallback {
         token_.transfer(address(TRSRY), amount_);
     }
 
+    /// @notice Burn any gOHM defaulted using the Cooler instead of the Clearinghouse.
+    function burn() external {
+        uint256 gohmBalance = gohm.balanceOf(address(this));
+
+        // Unstake and burn gOHM holdings.
+        gohm.approve(address(staking), gohmBalance);
+        MINTR.burnOhm(address(this), staking.unstake(address(this), gohmBalance, false, false));
```

Comment:
Doesn't the MINTR also need approval?

Reply:
since `MINTR` is trusted, we do an infinite approval when configuring the dependencies.

Reply:
Wait.

If that's the case, then https://github.com/sherlock-audit/2023-08-cooler-judging/issues/176 is invalid.

Why was this never brought up?

Reply:
Oh, I misunderstood. The infinite approval is in a new change. Nevermind.